test/tables.mediawiki

```diff
@@ -1,145 +1,152 @@
 Simple table with caption:
 
-{|
+{| class="wikitable"
 |+ Demonstration of simple table syntax.
-!align="right"| Right
-! Left
-!align="center"| Center
+|-
+! style="text-align: right;"| Right
+! style="text-align: left;"| Left
+! style="text-align: center;"| Center
 ! Default
 |-
-|align="right"| 12
-| 12
-|align="center"| 12
+| style="text-align: right;"| 12
+| style="text-align: left;"| 12
```

Comment:
Can we leave off a style tag if it's default-aligned?

Reply:
I could do that. However, this might be a bit messy, as the default lay-out (defined by the `class=wikitable`) gives headers (starting with `!`) a default center-align, and cells (starting with `|`) a default left-align, so I should do a switch on this to determine what is the default. (See also the [wikitable css page](https://en.wikipedia.org/wiki/Wikipedia:Catalogue_of_CSS_classes/Wikitable), which even seems to state that the default align for the caption depends on whether you view it on mobile or desktop.)

For now I went with the easy solution of "When an align is given I add it and when it is default I ignore it" (I'm not that much of a front-end/accessibility guy).

I'll see what I can do to implement this kind of switch.

Reply:
Furthermore, the default depends on whether the language is rtl or ltr, I don't see a way to infer what writing direction the language will be. Any tips on this?

Reply:
Yes, it sounds as if what you're doing is what I meant: if we have AlignDefault, then don't print any special alignment.

I'm just surprised by the difference in the tests -- I had assumed that the reason the alignments were left off before was that we had AlignDefault -- but maybe the old code just treated AlignLeft as AlignDefault?

Reply:
Yes it did